services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala

```diff
@@ -1,18 +1,35 @@
 package cromwell.pipeline.service
 
 import cromwell.pipeline.datastorage.dao.repository.DocumentRepository
+import cromwell.pipeline.datastorage.dto.project.configuration.ProjectConfigurationEntity
 import cromwell.pipeline.datastorage.dto.{ ProjectConfiguration, ProjectId }
 
 import scala.concurrent.{ ExecutionContext, Future }
 
 class ProjectConfigurationService(repository: DocumentRepository)(implicit ec: ExecutionContext) {
   import ProjectConfiguration._
 
-  def addConfiguration(projectConfiguration: ProjectConfiguration): Future[String] =
+  def addConfiguration(configuration: ProjectConfigurationEntity): Future[String] =
     repository
-      .updateOne(toDocument(projectConfiguration), "projectId", projectConfiguration.projectId.value)
+      .replaceOne(
+        toDocument(ProjectConfiguration(configuration)),
+        "projectId",
+        configuration.projectId.value
+      )
       .map(_.toString)
 
-  def getById(projectId: ProjectId): Future[Option[ProjectConfiguration]] =
-    repository.getByParam("projectId", projectId.value).map(_.headOption.map(fromDocument))
+  def getById(projectId: ProjectId): Future[Option[ProjectConfigurationEntity]] =
+    repository.getByParam("projectId", projectId.value).map { optDocument =>
+      optDocument.filter(document => fromDocument(document).isActive).map { document =>
+        val config = fromDocument(document)
+        ProjectConfigurationEntity(config.projectId, config.projectFileConfigurations)
+      }
+    }
+
+  def deactivateConfiguration(projectId: ProjectId): Future[String] =
+    getById(projectId).flatMap {
+      case Some(_) =>
+        repository.updateOneField("projectId", projectId.value, "isActive", false).map(_.toString)
```

Comment:
This `.toString` is a real headache. I guess I should create another task to get rid of it.